mlflow/utils/_capture_modules.py

```diff
@@ -83,16 +83,13 @@ def main():
     # Mirror `sys.path` of the parent process
     sys.path = json.loads(args.sys_path)
 
+    spark = None
     if flavor == mlflow.spark.FLAVOR_NAME and is_in_databricks_runtime():
-        try:
-            # pylint: disable=import-error
-            from dbruntime.spark_connection import initialize_spark_connection
-
-            initialize_spark_connection()
-        except Exception as e:
-            raise Exception(
-                "Attempted to initialize a spark session to load the spark model, but failed"
-            ) from e
+        # Clear 'PYSPARK_GATEWAY_PORT' and 'PYSPARK_GATEWAY_SECRET' to enforce launching a new JVM
+        # gateway
+        os.environ.pop("PYSPARK_GATEWAY_PORT", None)
+        os.environ.pop("PYSPARK_GATEWAY_SECRET", None)
+        os.environ["SPARK_DIST_CLASSPATH"] = "/databricks/jars/*"
```

Comment:
@Ngone51 @HyukjinKwon Could you help review this part of the PR ?

Context:
The code here will be run in databricks environment, and it is in a **subprocess** of pyspark driver process.
In the subprocess, later it will create a **local spark session**, the 3 lines here is doing some preparation work:

(1)
```
        os.environ.pop("PYSPARK_GATEWAY_PORT", None)
        os.environ.pop("PYSPARK_GATEWAY_SECRET", None)
```
this is for cleaning gateway env variables inherits from parent process, otherwise the py4j will connect to the existing driver py4j gateway.

(2)
```
os.environ["SPARK_DIST_CLASSPATH"] = "/databricks/jars/*"
```
This line is because, databricks runtime put all runtime jars in `/databricks/jars/*`
and `SPARK_DIST_CLASSPATH` is for setting additional jars path https://github.com/databricks/runtime/blob/d8d2022d1e0a57644a71ff930ffbdc5528d24fca/bin/spark-class#L54

Reply:
wow complicated .. Yes, setting `PYSPARK_GATEWAY_PORT` and `PYSPARK_GATEWAY_SECRET` to `None` would work here, this part loos fine. But I wonder if we can just simply stick to using threads instead of processes. That would make our lives easier ..

Reply:
@HyukjinKwon
Clarification: this PR is not related to the reuse py4j gateway server issue.
We create subprocess for capturing python package dependencies.
and here we create spark session of "local mode", which has nothing related to the existing py4j gateway server.

Reply:
LGTM